### ISSUE TITLE
rancher operator is created rancher-operator-system which is not installed in System project 

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#32898]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

- Deploy rancher HA 2.5.8
- The rancher operator chart/pod is seen deployed in rancher-operator-system which is not installed in System project of the local cluster

**Expected Result:**
Rancher operator system namespace should be in thhe System project of the local cluster

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.8
- Installation option (single install/HA): HA

**Comment by jakefhyde** [comment]
Discussion
-
Per discussion, we've defined the following requirements:

- [ ] Fix the `DefaultNamespaceAssigned` and `SystemNamespacesAssigned` conditions not being properly set for system and default project namespace assignment
- [ ] Add the `ensureNamespacesAssigned` check to project updated (pending successful condition set update)
- [ ] Unset the `SystemNamespacesAssigned` conditions on controller registration, which will happen on startup and election ensuring successful update

Notes
-
- Projects are updated on user interaction, as well as cluster sync
- Decided to only move namespaces with no `projectID` annotations (i.e. not currently associated with a project) that should be in the System project (i.e. defined in the `system-namespaces` setting) into the System project
- Only unset the `SystemNamespacesAssigned` on registration, as the Default project only requires the `default` namespace, and is unlikely to change
- The fix to add fleet auto-generated namespaces into the system-namespaces project is 2.6 onward, as is the hook to check if specific updates are required during an upgrade based on the current rancher version

**Comment by cbron** [comment]
Moving this to 2.5.11, don't merge into 2.5 until 2.5.10 goes out.

**Comment by jmcsagdc** [comment]
Verified in v2.5-e014cd0b24c798950cdbcd7d99d4fd83ea529d36-head

1. launch something older. I used 2.5.10
2. add a namespace outside of any project. I created "david"
3. use the api to add Harry as a System project
4. upgrade Rancher to 2.5-head
5. log in and see where david ended up in the Projects/Namespaces

EXPECTED:
david should be in the System project under the local cluster

![postupgrade2 5](https://user-images.githubusercontent.com/15971368/143966347-d1f9fb2a-5be5-40d0-82b4-d61f426febad.jpg)
